File: code_samples/csharp/error_handling_custom_exceptions.cs
using System;

/// <summary>
/// Error Handling: Custom Exceptions
/// AI/ML Training Sample
/// </summary>
public class CustomExceptions
{
    private string data;
    
    public CustomExceptions()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new CustomExceptions();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/dart/error_handling_custom_exceptions.dart
// Error Handling: Custom Exceptions
// AI/ML Training Sample

class CustomExceptions {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = CustomExceptions();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
